classes/external/enqueue_merge_request.php

```diff
@@ -0,0 +1,120 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+/**
+ * Version information
+ *
+ * @package     tool
+ * @subpackage  mergeusers
+ * @author      Nicola Vallinoto, Liguria Digitale
+ * @author      Jordi Pujol-Ahulló, SREd, Universitat Rovira i Virgili
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+namespace tool_mergeusers\external;
+defined('MOODLE_INTERNAL') || die();
+global $CFG;
+require_once("{$CFG->libdir}/externallib.php");
+defined('MOODLE_INTERNAL') || die();
+use \tool_mergeusers\merge_request;
+use external_function_parameters;
+use external_multiple_structure;
+use external_single_structure;
+use external_value;
+class enqueue_merge_request extends \external_api {
+    /**
+     * Webservice enqueue_merge_request parameters
+     *
+     * @return external_function_parameters
+     */
+    public static function execute_parameters(): external_function_parameters {
+            return new external_function_parameters(
+            [
+                'removeuserfield' => new external_value(PARAM_TEXT, 'Remove user field'),
+                'removeuservalue' => new external_value(PARAM_RAW, 'Remove user value'),
+                'keepuserfield' => new external_value(PARAM_TEXT, 'Keep user field'),
+                'keepuservalue' => new external_value(PARAM_RAW, 'Keep user value'),
+            ]
+        );
+    }
+    /**
+     * Return identifier of the enqueue merge request
+     *
+     * @param $removeuserid
```

Comment:
see how phpdoc works. Mainly: All these lines need the type of the parameter like this `@param {type} $paramname {description}`

Reply:
done